src/suggest.c

```diff
@@ -177,6 +177,10 @@ int RSSuggestDelCommand(RedisModuleCtx *ctx, RedisModuleString **argv, int argc)
   const char *str = RedisModule_StringPtrLen(argv[2], &len);
   RedisModule_ReplyWithLongLong(ctx, Trie_Delete(tree, str, len));
 
+  if (tree->size == 0) {
```

Comment:
Consider moving to a different PR (changing the suggestion is more "breaking" since it is a redis key)

Reply:
OK. The changes to suggestion dictionaries are part of a differente PR: https://github.com/RediSearch/RediSearch/pull/5320